### PR TITLE
Cleanup `/test/rocmlir-driver/misc_e2e`

### PR DESCRIPTION
**Pull request by zhanglx13**
This PR cleans up the `test/rocmlir-driver/misc_e2e` dir

- `conv2d_harness_cpu_verifier_kyxc_nhwc_nhwk.mlir` and `conv2d_harness_cpu_verifier_yxck_nhwc_nhwk.mlir` are moved into `test/rocmlir-driver/conv2d_harness/cpu_verifier/`
- The test in `conv2d_harness_cpu_verifier_kyxc_nhwc_nhwk_f16.mlir` is added into `test/e2e/conv2d_regression_fwd.toml`.
- Tests in `conv2d_host_validation.mlir` and `conv2d_host_validation_f16.mlir` (both `RUN` and `FIXME`) are added into `test/e2e/Resnet101Config.toml`, which is enabled as other ROCK E2E tests by `ROCK_E2E_TEST_ENABLED`.
- Tests in `conv2d_host_validation_random.mlir` are added into `test/e2e/conv2d_regression_fwd.toml`.
- Get rid of `ROCMLIR_DRIVER_MISC_E2E_TEST_ENABLED`
- Delete `misc_e2e` dir.

Note that some tests in `misc_e2e` have `-rand_type float` as random inputs. This issue will be addressed in https://github.com/ROCmSoftwarePlatform/rocMLIR/pull/869.

### CODE REVIEW

File: mlir/test/e2e/CMakeLists.txt [hunk 1]
```diff
@@ -40,6 +40,11 @@ if (ROCK_E2E_TEST_ENABLED)
     conv2d_wrw_perf_config
   )
 endif()
+if (ROCMLIR_DRIVER_MISC_E2E_TEST_ENABLED)
```
Comment (krzysz00) [inline mlir/test/e2e/CMakeLists.txt]: I'd argue this is a good time to get rid of `ROCMLIR_DRIVER_MISC_E2E_TEST_ENABLED`

Comment (zhanglx13) [reply mlir/test/e2e/CMakeLists.txt]: Previously, these tests are only enabled with fixed inputs. I'll add them as regular e2e tests and see if they can pass with random inputs.

Comment (krzysz00) [reply mlir/test/e2e/CMakeLists.txt]: Yeah, worth checking. Could be this was an old bug or something.

**Review by krzysz00 — APPROVED** [review]
Seems fine to me

However, IIRC, there were some Resnext101 configs that were identical to Resnet50 and so got commented out. If those comments were in the files you deleted, can you preserve them in the config file?

**Comment by zhanglx13** [comment]
I did not touch any Resnext101 and Resnet50 tests. Some tests in misc_e2e dir are from Resnet101, which I assume is different from Resnext101.
The configs are commented out in Resnext101Config.toml if they also appear in Resnet50Config.toml.

**Comment by krzysz00** [comment]
Ah. I'm fantastic an noticing that sort of difference.

And so, without previous reservations, LGTM

**Pull request merged**